bootstrap/service.go

```diff
@@ -262,26 +286,112 @@ func (bs bootstrapService) UpdateConnections(ctx context.Context, token, id stri
 			return ErrThings
 		}
 	}
-	if err := bs.configs.UpdateConnections(ctx, owner, id, channels, connections); err != nil {
+	if err := bs.configs.UpdateConnections(ctx, user.GetDomainId(), id, channels, connections); err != nil {
 		return errors.Wrap(errUpdateConnections, err)
 	}
 	return nil
 }
 
+func (bs bootstrapService) listClientIDs(ctx context.Context, userID, permission string) ([]string, error) {
+	tids, err := bs.auth.ListAllObjects(ctx, &magistrala.ListObjectsReq{
+		SubjectType: auth.UserType,
+		Subject:     userID,
+		Permission:  permission,
+		ObjectType:  auth.ThingType,
+	})
+	if err != nil {
+		return nil, errors.Wrap(svcerr.ErrNotFound, err)
+	}
+	return tids.Policies, nil
+}
+
+func (bs bootstrapService) filterAllowedThingIDs(ctx context.Context, userID, permission string, thingIDs []string) ([]string, error) {
+	var ids []string
+	tids, err := bs.auth.ListAllObjects(ctx, &magistrala.ListObjectsReq{
+		SubjectType: auth.UserType,
+		Subject:     userID,
+		Permission:  permission,
+		ObjectType:  auth.ThingType,
+	})
+	if err != nil {
+		return nil, errors.Wrap(svcerr.ErrNotFound, err)
+	}
+	for _, thingID := range thingIDs {
+		for _, tid := range tids.Policies {
+			if thingID == tid {
+				ids = append(ids, thingID)
+			}
+		}
+	}
+	return ids, nil
+}
+
+func (bs bootstrapService) checkSuperAdmin(ctx context.Context, userID string) error {
+	res, err := bs.auth.Authorize(ctx, &magistrala.AuthorizeReq{
+		SubjectType: auth.UserType,
+		Subject:     userID,
+		Permission:  auth.AdminPermission,
+		ObjectType:  auth.PlatformType,
+		Object:      auth.MagistralaObject,
+	})
+	if err != nil {
+		return err
+	}
+	if !res.Authorized {
+		return svcerr.ErrAuthorization
+	}
+	return nil
+}
+
 func (bs bootstrapService) List(ctx context.Context, token string, filter Filter, offset, limit uint64) (ConfigsPage, error) {
-	owner, err := bs.identify(ctx, token)
+	user, err := bs.identify(ctx, token)
 	if err != nil {
 		return ConfigsPage{}, errors.Wrap(svcerr.ErrAuthentication, err)
 	}
-	return bs.configs.RetrieveAll(ctx, owner, filter, offset, limit), nil
+
+	if err := bs.checkSuperAdmin(ctx, user.GetId()); err == nil {
+		return bs.configs.RetrieveAll(ctx, user.GetDomainId(), "", filter, offset, limit), nil
+	}
+
+	// Check user is admin of domain, if yes then show listing on domain context.
+	_, authErr := bs.authorize(ctx, "", auth.UsersKind, user.GetId(), auth.AdminPermission, auth.DomainType, user.GetDomainId())
+	if authErr == nil {
+		return bs.configs.RetrieveAll(ctx, user.GetDomainId(), "", filter, offset, limit), nil
+	}
+
+	rtids, err := bs.listClientIDs(ctx, user.GetId(), auth.ViewPermission)
```

Comment:
```suggestion
	rtids, err := bs.listClientIDs(ctx, user.GetId())
```
Since we are using it at one place